dashboard/app/controllers/datablock_storage_controller.rb

```diff
@@ -0,0 +1,347 @@
+# DatablockStorage is the backend for the 'Data' tab of Applab, the 'Data' blocks
+# and the data browser. DatablockStorage stores student-controlled data generated
+# by and with the data blocks like `createRecord`, `updateRecord`, `setKeyValue`, etc.
+#
+# These backend methods are accessed exclusively by a thin wrapper on the JS side:
+# datablockStorage.js. Each method in this controller has a corresponding method there.
+#
+# Methods in this controller are available as (see `:datablock_storage` in routes.rb):
+# /datablock_storage/:channel_id/:method_name
+#
+# Student data is stored in MySQL in corresponding ActiveRecord models:
+# - datablock_storage_table.rb: stores a list of tables and their columns, after this file
+#   this model contains the most code.
+# - datablock_storage_record.rb: stores the records in each table
+# - datablock_storage_kvp.rb: stores key-value pairs
+#
+# Metadata for the code.org defined datasets are stored in:
+# - datablock_storage_library_manifest.rb
+#
+# Methods are broken into sections, for example the Key-Value-Pair API, Table API,
+# Table Column API, Table Record API, Library Manifest API & Project API
+#
+# More details can be found in the PR that initially created Datablock Storage:
+# https://github.com/code-dot-org/code-dot-org/pull/56279
```

Comment:
As I was reading through and trying to learn about the data model, the main question in my head was "How are we doing both a KV store and a more traditional table (Relational DB?) at the same time?"

I couldn't find much about the KV store in the documentation (comments, issues, or PR description). In retrospect, this is probably because it's actually the much simpler of the two. But when I was learning, I assumed it would be the more complex because we're using mySql as our underlying DB and in my mind that's more similar to the tables students are making than to the KV stores they're making.

It would be really helpful to have a bit more description of the KV store that helps me know I shouldn't overthink it and it's pretty straightforward. A few main possibilities that come to mind are in the PR description, the header of this file, or in the `Key-Value-Pair API` section below, as those are where I looked for details before tracing through the code.

Reply:
I think the best place to look to answer that would be in the model file for the key-value pairs model (datablock_storage_kvp.rb). This file describes the schema of the table where we're storing the data & implementation of each of the methods we're calling here on the model, & is guaranteed to not be out of date. It sounds like you found your way to the issue describing the models, but it was out of date and caused some confusion with channel_id vs project_id; But you eventually did find your way to the model file and find the answers you were looking for. Do you have any suggestions on what we could say here to lead you in the direction of looking at the model files first?